YapDatabase/YapDatabaseTransaction.m

```diff
@@ -2726,9 +2004,17 @@ - (void)_enumerateKeysInCollection:(NSString *)collection
 	YapDatabaseString _collection; MakeYapDatabaseString(&_collection, collection);
 	sqlite3_bind_text(statement, bind_idx_collection, _collection.str, _collection.length, SQLITE_STATIC);
 	
-	int status;
-	while ((status = sqlite3_step(statement)) == SQLITE_ROW)
-	{
+    YapDatabaseConnection *connection = self.connection;
+    __block int status;
+    [self whileLoopWithBatchSize:kDefaultBatchSize
+                       condition:^{
+                           if (stop || mutation.isMutated) {
+                               return NO;
+                           }
+                           
+                           status = sqlite3_step(statement);
+                           return (BOOL) (status == SQLITE_ROW);
+                       } loopBlock:^{
```

Comment:
To keep the diffs readable, I didn't bother re-indenting the contents of these loops.